packages/labs/analyzer/src/lib/model.ts

```diff
@@ -304,33 +403,64 @@ export type ClassHeritage = {
 export interface ClassDeclarationInit extends DeclarationInit {
   node: ts.ClassDeclaration;
   getHeritage: () => ClassHeritage;
+  fieldMap?: Map<string, ClassField> | undefined;
```

Comment:
Can the key be `PropertyKey` instead of `string`? Do we support, for example, symbols here?

Reply:
Hmm, I guess we should, at least in the analyzer?

We'd need to come up with a syntax for emitting that in CEM though, since there's no schema space to describe a Symbol name. Maybe class member could have `name: string | Reference`, where if it's a reference it points to a symbol? @justinfagnani

Reply:
Will handle separate from this PR. I added an issue on CEM schema here: 

* https://github.com/webcomponents/custom-elements-manifest/issues/114

and an analyzer issue here: 

* https://github.com/lit/lit/issues/3565

Reply:
Do we have real-world cases where we want to analyze symbol-keyed fields? Just wondering if we can punt for a while.

Since you can't bind to them in lit-html, they'd only be good for imperative usage too, which TypeScript should already handle just fine.

Reply:
Yeah, we can definitely punt for a while, seems pretty low-priority.